controllers/storagecluster/storageclient.go

```diff
@@ -98,10 +98,15 @@ func (s *storageClient) ensureDeleted(r *StorageClusterReconciler, storagecluste
 	}
 	storageClient := &ocsclientv1a1.StorageClient{}
 	storageClient.Name = storagecluster.Name
+	if err := r.Get(r.ctx, client.ObjectKeyFromObject(storageClient), storageClient); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get storageclient %s: %v", storageClient.Name, err)
+		}
+		return ctrl.Result{}, nil
+	}
```

Comment:
this err handling is quite confusing, what bug is being solved?

Reply:
We are removing the finalizer from storageclient after deleting it. If we don't load the storageclient, the finalizer won't be present in the local object and it won't go through the update operation to remove the finalizer on the object

Reply:
maybe this can be restructured as
```
if err := r.get(); client.IgnoreNotFound(err) != nil { handle api-error }
else if storageClient.UID == "" { nothing to do here }
// rest as usual + no check for not found in delete
```
wdyt?